### PR TITLE
Long awaited clean up towards launch

### LINKED ISSUE #196: Implement contracts that can be settled using our internal oracle service
#### User Story

- As an <audience/user>:  dev
- I want to <action/task>: be able to have contracts that are settled using an internal oracle
- so that <outcome/goal/benefit>: we can gradually move to more decentralized solutions

#### Type

- [x] Enhancement
- [ ] Maintenance
- [ ] Refactor

#### Description
Create implementation of base MarketContract.sol 

#### Definition of Done
- [ ] new MarketContract.sol implementation
- [ ] testing!

### PR DESCRIPTION
**Pull request by pelsasser**
##### Description
- updated solidity to 0.4.25
- updates truffle to v4.1.15
- updates open zeppelin to 2.0.1
- removes all non essential code for mainnet
- adds delay from settlement to withdraw of funds 
- creates new contract that can be settled using our internal oracle service
- adds in ability for us to pass in longTokenSymbol and shortTokenSymbol into constructor.
- adds ability for us to arbitrate the settlement.  We can now delay settlement, override settlement, and push a contract into early settlement.
- fixes migrations

##### Refers/Fixes
#196 #188 #156 #199 #200

### CODE REVIEW

**Comment by pelsasser** [comment]
@perfectmak can you review this when you are ready! Lots of changes.

Creating tickets for the remaining needed changes.

File: contracts/MarketContract.sol [hunk 1]
```diff
@@ -72,20 +79,36 @@ contract MarketContract is Creatable {
...
         COLLATERAL_PER_UNIT = MathLib.calculateTotalCollateral(PRICE_FLOOR, PRICE_CAP, QTY_MULTIPLIER);
+        FEE_PER_UNIT = MathLib.calculateFeePerUnit(PRICE_FLOOR, PRICE_CAP, QTY_MULTIPLIER, contractSpecs[4]);
+
+        // create long and short tokens
+        StringLib.slice memory pathSlice = contractNames.toSlice();
+        StringLib.slice memory delim = ",".toSlice();
+        require(pathSlice.count(delim) == 2, "ContractNames must contain 3 names");  //contractName,lTokenName,sTokenName
+        CONTRACT_NAME = pathSlice.split(delim).toString();
+
+        PositionToken longPosToken = new PositionToken("Long Position Token", pathSlice.split(delim).toString(), 0);
```
Comment (perfectmak) [inline contracts/MarketContract.sol]: I got confused in the whole pathSlice logic, we should probably add a test to MarketContracts to ensure the correct symbol is extracted for its positions tokens symbol.

What I want to ask here though is it okay to leave `Long Position Token` and `Short Position Token` as the name for all MPX Position tokens (symbols are different alright)?

Comment (perfectmak) [reply contracts/MarketContract.sol]: Also, it seems contract names have changed from what we have before, we should confirm in our other codebase that we are no more parsing names based on the old convention.

File: contracts/MarketCollateralPool.sol [hunk 2]
```diff
@@ -64,10 +74,17 @@ contract MarketCollateralPool is Ownable {
         require(!marketContract.isSettled(), "Contract is already settled");
 
         uint neededCollateral = MathLib.multiply(qtyToMint, marketContract.COLLATERAL_PER_UNIT());
+        uint feeAmount = MathLib.multiply(qtyToMint, marketContract.FEE_PER_UNIT());
+        uint totalTransferAmount = neededCollateral.add(feeAmount);
```
[the remark this answers is not in the document]
Comment (pelsasser) [reply contracts/MarketCollateralPool.sol]: Yup!

File: contracts/MarketContract.sol (hunk 1 above)
Comment (pelsasser) [reply contracts/MarketContract.sol]: I think so, I am debating making it `MARKET Protocol Long Position Token` so people know where it came from as well.

**Pull request merged**